### PR TITLE
AIRFLOW-3632 Adding replace_microseconds parameter to dag_runs endpoint

### PR DESCRIPTION
**Pull request by alpeshpandya**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3632
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Adding replace_microseconds parameter to dag_runs (trigger_dag) api allows user/scripts to submit run requests to Airflow without having to wait for one second for subsequent submission.
Example of submitting with replace_microseconds parameter set to No:
curl -X POST \
  http://localhost:8080/api/experimental/dags/example_bash_operator/dag_runs \
  -H 'Content-Type: application/json' \
  -H 'cache-control: no-cache' \
  -d '{"conf":"", "replace_microseconds":"N"}'

The Parameter is still optional so not passing is interpreted as replace_microseconds Yes following convention of underlying trigger_dag method.

### Tests

- [x] My PR does not need testing as there are no existing test cases for existing end points. The underlying trigger_dag method is not changed which is covered by existing test cases.

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by m-swat** [comment]
I 'm waiting for it!

File: airflow/www/api/experimental/endpoints.py [hunk 1]
```diff
@@ -74,8 +77,16 @@ def trigger_dag(dag_id):
 
             return response
 
+    # Default replace_microseconds to True follow convention from trigger.trigger_dag
+    replace_microseconds = True
```
Comment (zhongjiajie) [inline airflow/www/api/experimental/endpoints.py:81]: I think we could simplify it like below
```suggestion
    replace_microseconds = True
    if 'replace_microseconds' in data and data['replace_microseconds'].lower() not in YES_VALUES:
        replace_microseconds = False
```

File: airflow/www/api/experimental/endpoints.py [hunk 2]
```diff
@@ -37,6 +37,9 @@
 
 api_experimental = Blueprint('api_experimental', __name__)
 
+# Possible yes values to replace_microseconds parameter for dag_runs (trigger_dag) endpoint
+YES_VALUES = ['true', '1', 't', 'y', 'yes']
```
Comment (zhongjiajie) [inline airflow/www/api/experimental/endpoints.py:41]: Re-order values index
```suggestion
YES_VALUES = ['true', 't', 'y', 'yes', '1']
```

**Review by zhongjiajie — COMMENTED** [review]
Maybe we could simplify the code

**Review by ashb — CHANGES_REQUESTED** [review]
Can you add some unit tests, as well as apply the suggestions zhongjiajie has suggested?

**Comment by chatnuere** [comment]
any news ?

this pull request will be very usefull to a lot of people 

I'm using a small hack for me but an official support would be great 

thx for your job @alpeshpandya

**Comment by aladinoss** [comment]
Completly Agree, this PR is very usefull.

**Comment by ashb** [comment]
Someone is welcome to take up this PR, add unit tests and address the comments.

**Pull request closed without merging**